### ISSUE TITLE
Correlation in Azure Application Insights is not working

### ISSUE THREAD

**Issue by gverstraete** [arcus-azure/arcus.observability#69]
**Describe the bug**
in my code I added the new packages for correlation and the ones for logging. But when combining the two i see all the requests seperately from each other and not nested.

![image](https://user-images.githubusercontent.com/31242865/78107365-9c9df200-73f5-11ea-9fe4-6d09695b0101.png)

**To Reproduce**
So this is my Program:
```csharp
public class Program
    {
        public static void Main(string[] args)
        {
            CreateHostBuilder(args).Build().Run();
        }

        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureWebHostDefaults(webBuilder =>
                {
                    webBuilder.UseSerilog().UseStartup<Startup>();
                });
    }
```

And the startup:
```csharp
    public class Startup
    {
        public void ConfigureServices(IServiceCollection services)
        {
            IServiceCollectionExtensions.AddCorrelation(services);

            services.AddMvc(options => options.EnableEndpointRouting = false).SetCompatibilityVersion(CompatibilityVersion.Version_3_0); ;

            services.AddSingleton<ITestRepo, TestRepo>();

            services.AddHttpCorrelation();

            services.AddSwaggerGen(c =>
            {
                c.SwaggerDoc("Api", new Microsoft.OpenApi.Models.OpenApiInfo { Title = "API", Version = "#{VERSION}#" });
            });
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }

            app.UseSwagger();
            app.UseSwaggerUI(c => { c.SwaggerEndpoint("/swagger/Api/swagger.json", "BMID.Fraud.Onboarding.Api"); });

            app.UseMiddleware<RequestLoggingMiddleware>();


            app.UseHttpsRedirection();
            app.UseMvc();

            app.UseHttpCorrelation();

            Log.Logger = new LoggerConfiguration()
                .MinimumLevel.Verbose()
                .Enrich.WithCorrelationInfo()
                .WriteTo.AzureApplicationInsights("xxxxx")
                .WriteTo.Console(outputTemplate:"{Message}")
                .CreateLogger();

        }
```

**Expected behavior**
I would expect to see the correlation in action :)

**Additional context**
- .NET Core 3.1
- VS 2019

```xml
  <ItemGroup>
    <PackageReference Include="Arcus.Observability.Correlation" Version="0.1.0-preview-5" />
    <PackageReference Include="Arcus.Observability.Telemetry.Core" Version="0.1.0-preview-5" />
    <PackageReference Include="Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights" Version="0.1.0-preview-5" />
    <PackageReference Include="Arcus.WebApi.Correlation" Version="1.0.0-preview-1" />
    <PackageReference Include="Arcus.WebApi.Logging" Version="1.0.0-preview-1" />
    <PackageReference Include="Dapper" Version="2.0.30" />
    <PackageReference Include="Dapper.Contrib" Version="2.0.30" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.0.0" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="3.0.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="3.0.0" />
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="3.0.0" />
    <PackageReference Include="Serilog.AspNetCore" Version="3.2.0" />
    <PackageReference Include="Swashbuckle.AspNetCore.Swagger" Version="5.2.1" />
    <PackageReference Include="Swashbuckle.AspNetCore.SwaggerGen" Version="5.2.1" />
    <PackageReference Include="Swashbuckle.AspNetCore.SwaggerUI" Version="5.2.1" />
  </ItemGroup>

```

**Comment by stijnmoreels** [comment]
> But the console still doesn't show the correct rendered message. Even without using our extensions, placing the string directly...

The console couldn't render the context correctly, had to add this one:
`.Destructure.ByTransforming<Dictionary<string, object>>(dic => String.Join(", ", dic.Select(kv => $"[{kv.Key}, {kv.Value}]")))`

**Comment by tomkerkhove** [comment]
What does it look like then?

This might be something we have to fix as well, would you mind opening a dedicated issue please?

**Comment by stijnmoreels** [comment]
It doesn't look like anything. The message is not rendered bc of this.

**Comment by tomkerkhove** [comment]
It should just work out of the box with our extensions, so if you have to add `.Destructure.ByTransforming<Dictionary<string, object>>(dic => String.Join(", ", dic.Select(kv => $"[{kv.Key}, {kv.Value}]")))` to make it usable I'd argue we have to make it even simpler.

What was the configuration you had to use in the end?

**Comment by stijnmoreels** [comment]
> It should just work out of the box with our extensions, so if you have to add `.Destructure.ByTransforming<Dictionary<string, object>>(dic => String.Join(", ", dic.Select(kv => $"[{kv.Key}, {kv.Value}]")))` to make it usable I'd argue we have to make it even simpler.
> 
> What was the configuration you had to use in the end?

```csharp
new LoggerConfiguration()
  .Destructure.ByTransforming<Dictionary<string, object>>(dic => String.Join(", ", dic.Select(kv => $"[{kv.Key}, {kv.Value}]")))
  .Enrich.WithCorrelationInfo(new HttpCorrelationInfoAccessor(app.ApplicationServices.GetService<IHttpContextAccessor>()))
  .WriteTo.Console()
  .WriteTo.AzureApplicationInsights()
```

**Comment by tomkerkhove** [comment]
If that line is a must then we have to help as Arcus :)
We cause the issue, so have to fix the issue.

**Comment by tomkerkhove** [comment]
Did we have an open action for Observability here or only for Web API?

**Comment by tomkerkhove** [comment]
We checked and everything should be tackled. Are you missing something as part of this @gverstraete?

**Comment by tomkerkhove** [comment]
Correlation will be available in v0.1.0 which we're shipping now.

**Comment by gverstraete** [comment]
Thanks! Just checked and it worked; sorry for the late reply!
